os/se/ameba/security_ameba_wrapper_tz.c

```diff
@@ -104,24 +105,40 @@ typedef struct {
 } inout_struc;
 
 typedef struct {
-	uint32_t factorykey_device_idx;
-	uint32_t factorykey_cert_idx;
 	uint32_t factory_cert_addr;
 	uint32_t factory_key_addr;
 	uint32_t sstorage_addr;
 	uint32_t samsung_key_addr;
+	hal_key_type factory_slot_key_type[USABLE_FACTORY_KEY_INDEX];
 } factory_struc;
 
 static uint8_t *ns_buf = NULL;
-
 extern void *rtl_set_ns_func(void);
 
+/**
+ * Helper function
+ */
+static int se_ameba_index_chk(uint32_t key_idx)
+{
+	uint32_t ret_key_idx = HAL_SUCCESS;
+	if (key_idx >= TOTAL_KEY_STORAGE_INDEX_MAX) {	/* Index Out of Range */
+		ret_key_idx = HAL_INVALID_SLOT_RANGE;
+	}
+
+	/* Check if in valid Factory range */
+	if ((key_idx >= USABLE_FACTORY_KEY_INDEX) && (key_idx < FACTORY_KEY_INDEX_MAX)) {
+		ret_key_idx = HAL_INVALID_SLOT_RANGE;
+	}
+
+	return ret_key_idx;
```

Comment:
`ret_key_idx` is unsigned int type, but the function return type is signed int. Please match the type.

Reply:
Done, return type is change to int.